### PR TITLE
Picking appointment timeslot

### LINKED ISSUE #25: Add time picker to the book appointment page

### PR DESCRIPTION
**Pull request by bl-nero**
Closes #25.

To test it, run `npm start` and point your browser to `http://localhost:3000/hospitals/1/book-appointment`.

The layout is very basic right now, but the front-end part of functionality is working.

### CODE REVIEW

File: src/components/AppointmentScheduler.tsx [hunk 1]
```diff
@@ -1,47 +1,92 @@
 import React, { useState, useEffect } from "react";
```
Comment (kaashmonee) [inline src/components/AppointmentScheduler.tsx:1]: Please consider adding documentation to this file. Maybe a quick docstring-esque description to the subroutines defined and called in here?

File: src/components/AppointmentTimePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import React, { ChangeEvent } from "react";
```
Comment (kaashmonee) [inline src/components/AppointmentTimePicker.tsx:1]: Please consider adding documentation to this file as well.

File: src/components/AppointmentScheduler.test.tsx [hunk 3]
```diff
@@ -1,27 +1,79 @@
 import React from "react";
```
Comment (kaashmonee) [inline src/components/AppointmentScheduler.test.tsx:1]: How can I run these tests?

File: src/components/AppointmentDatePicker.test.tsx [hunk 4]
```diff
@@ -1,7 +1,8 @@
 import React from "react";
```
Comment (kaashmonee) [inline src/components/AppointmentDatePicker.test.tsx:1]: How can I run these tests?

File: src/components/AppointmentTimePicker.test.tsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
```
Comment (kaashmonee) [inline src/components/AppointmentTimePicker.test.tsx:1]: How to run these tests?

**Comment by bl-nero** [comment]
Thank you for comments, @kaashmonee ! I'll address them later today.

File: src/components/AppointmentScheduler.test.tsx (hunk 3 above)
Comment (bl-nero) [reply src/components/AppointmentScheduler.test.tsx:1]: Run `npm install` and then `npm test`. See the readme file for details.

File: src/components/AppointmentDatePicker.test.tsx (hunk 4 above)
Comment (bl-nero) [reply src/components/AppointmentDatePicker.test.tsx:1]: Run `npm install` and then `npm test`. See the readme file for details.

File: src/components/AppointmentTimePicker.test.tsx (hunk 5 above)
Comment (bl-nero) [reply src/components/AppointmentTimePicker.test.tsx:1]: Run `npm install` and then `npm test`. See the readme file for details.

**Comment by bl-nero** [comment]
I updated the PR description, will deal with code comments later today.

**Comment by bl-nero** [comment]
@kaashmonee PTAL

**Comment by kaashmonee** [comment]
@bl-nero pointing my browser to `localhost:3000/hospitals/1/book-appointment` didn't result in any changes on the frontend. What should I expect to happen here?

File: src/components/AppointmentTimePicker.test.tsx (hunk 5 above)
Comment (kaashmonee) [reply src/components/AppointmentTimePicker.test.tsx:1]: Thanks, I was able to run the tests successfully.

**Comment by bl-nero** [comment]
@kaashmonee Here's how it should look like (on a mobile screen):
calendar
timeslot

**Comment by bl-nero** [comment]
@kaashmonee May I ask for approval?

**Comment by kaashmonee** [comment]
@bl-nero, yep, it just seems that there are merge conflicts at the moment.

**Comment by bl-nero** [comment]
@kaashmonee Technically, you can still approve changes in GitHub even if there are merge conflicts. I'll treat your word as an approval, anyway, and merge it.

**Pull request merged**